napalm-ruckus-fastiron/fi.py

```diff
@@ -0,0 +1,1378 @@
+        """Finds a word of a long_string and returns the value in the nth position"""
+        count = 0                           # counter
+        split_string = long_string.split()  # breaks long string into string of substring
+        values = []                         # creates a list
+        for m in split_string:              # goes through substrings one by one
+            count += 1                      # increments counter
+            if m == word:                   # if substring and word match then specific value
+                values.append(split_string[count + pos])    # is added to list that is returned
+        return values
+
+    @staticmethod
+    def find_words(output, word_list, pos_list):
+        """   """
+        dictionary = {}
+        if len(word_list) != len(pos_list):             # checks word, pos pair exist
+            return None
+
+        if len(word_list) == 0 or len(pos_list) == 0:   # returns NONE if list is empty
+            return None
+
+        size = len(word_list)
+        sentence = output.split()                   # breaks long string into separate strings
+
+        for m in range(0, size):                    # Iterates through size of word list
+            pos = int(pos_list.pop())               # pops element position and word pair in list
+            word = word_list.pop()
+            if sentence.__contains__(word):         # checks if word is contained in text
+                indx = sentence.index(word)         # records the index of word
+                dictionary[word] = sentence[indx + pos]
+
+        return dictionary
+
+    @staticmethod
+    def creates_list_of_nlines(my_string):
+        """ Breaks a long string into separated substring"""
+        temp = ""                       # sets empty string, will add char respectively
+        my_list = list()                # creates list
+        for val in range(0, len(my_string)):    # iterates through the length of input
+            if my_string[val] == '\n' and temp == "":
+                continue
+            elif my_string[val] == '\n' or val >= len(my_string):    # add what was found
+                my_list.append(temp)
+                temp = ""
+            else:
+                if my_string[val] == ' ' and my_string[val+1] == '\n':
+                    continue
+                temp += my_string[val]
+
+        return my_list
+
+    @staticmethod
+    def delete_if_contains(nline_list, del_word):
+        temp_list = list()                          # Creates a list to store variables
+        for a_string in nline_list:                 # iterates through list
+            if a_string .__contains__(del_word):    # if word matches, word is skipped
+                continue
+            else:
+                temp_list.append(a_string.split())  # Word didn't match store in list
+        return temp_list
+
+    @staticmethod
+    def facts_uptime(my_string):  # TODO check for hours its missing....

```

Comment:
Perhaps possible to just call `napalm.ios.IOSDriver.parse_uptime` ?
https://github.com/napalm-automation/napalm/blob/develop/napalm/ios/ios.py#L901

Reply:
I looked into there implementation and this wont be possible without some changes in there function.
Reason:
they  ".split()" the string only after a "," is found and assume that the first value after the comma (split) will be a numerical value.  This is due to the ways there "show version" data is structured.  We do not have the same structure :(

My current implementation finds the time keywords (seconds, minutes, hours, days...) and grabs the value before it as they are pre-fix to the keywords.

I will revise this portion and make the code more readable :). This reminds me i should look at not reinventing the wheel. Thank you for the great suggestion!